generals/envs/gymnasium_generals.py

```diff
@@ -102,6 +100,7 @@ def step(self, action: Action) -> tuple[Observation, SupportsFloat, bool, bool,
         actions = {self.agent_id: action, self.npc.id: npc_action}
 
         observations, infos = self.game.step(actions)
+        # From observations, pick only stuff for the agent
```

Comment:
"From all observations, pick only those relevant for the given agent" sounds better to me